com.unity.ml-agents/Runtime/Actuators/ActionSpec.cs

```diff
@@ -20,23 +30,23 @@ namespace Unity.MLAgents.Actuators
         ///
         /// For an IActuator with a Continuous it will be null.
         /// </summary>
-        public readonly int[] BranchSizes;
+        public int[] BranchSizes { get { return m_BranchSizes; } set { m_BranchSizes = value; } }
 
         /// <summary>
         /// The number of actions for a Continuous <see cref="SpaceType"/>.
         /// </summary>
-        public int NumContinuousActions { get; }
+        public int NumContinuousActions { get { return m_NumContinuousActions; } set { m_NumContinuousActions = value; } }
 
         /// <summary>
         /// The number of branches for a Discrete <see cref="SpaceType"/>.
         /// </summary>
-        public int NumDiscreteActions { get; }
+        public int NumDiscreteActions { get { return m_NumDiscreteActions; } set { m_NumDiscreteActions = value; } }
 
         /// <summary>
         /// Get the total number of Discrete Actions that can be taken by calculating the Sum
         /// of all of the Discrete Action branch sizes.
         /// </summary>
-        public int SumOfDiscreteBranchSizes { get; }
+        public int SumOfDiscreteBranchSizes { get { return m_SumOfDiscreteBranchSizes; } set { m_SumOfDiscreteBranchSizes = value; } }
```

Comment:
Probably out of scope, but why is this public and not internal? I feel like ActionSpec should be only nContinuous and DiscreteBranchSizes...

Reply:
Removed m_SumOfDiscreteBranchSizes and m_NumDiscreteActions, and derive them from branch sizes.